src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -624,63 +606,54 @@ def assignJobLocations(self):
           - SE name of the site to run at
         """
         jobsToSubmit = {}
-        jobsToUncache = []
         jobsCount = 0
         exitLoop = False
         jobSubmitLogBySites = defaultdict(Counter)
         jobSubmitLogByPriority = defaultdict(Counter)
 
         # iterate over jobs from the highest to the lowest prio
-        for jobPrio in sorted(self.cachedJobs, reverse=True):
+        for jobPrio in sorted(self.jobsByPrio, reverse=True):
 
             # then we're completely done and have our basket full of jobs to submit
             if exitLoop:
                 break
 
-            # start eating through the elder jobs first
-            for job in sorted(self.cachedJobs[jobPrio].values(), key=itemgetter('timestamp')):
-                jobid = job['id']
-                jobType = job['type']
+            # can we assume jobid=1 is older than jobid=3? I think so...
```

Comment:
Does python return keys in order. (I think it is)?

Reply:
From what I read, sets are unordered, even though it returns ordered data in my interactive tests. To be safe, I sort them in the line below.